egui_extras/src/image.rs

```diff
@@ -21,6 +35,36 @@ impl RetainedImage {
             size: image.size,
             image: Mutex::new(image),
             texture: Default::default(),
+            filter: Default::default(),
+        }
+    }
+
+    /// Set the texture filter to use with the image
+    ///
+    /// This should be called immediately after constructing the [`RetainedImage`] and will return
+    /// an error if it is called after the image has already been uploaded to the GPU.
+    ///
+    /// For example:
+    ///
+    /// ```rust
+    /// # use egui_extras::RetainedImage;
+    /// # use egui::{Color32, epaint::{ColorImage, textures::TextureFilter}};
+    /// # let pixels = vec![Color32::BLACK];
+    /// # let color_image = ColorImage {
+    /// #   size: [1, 1],
+    /// #   pixels,
+    /// # };
+    /// #
+    /// let image = RetainedImage::from_color_image("my_image", color_image)
+    ///     .with_texture_filter(TextureFilter::Nearest)
+    ///     .unwrap();
```

Comment:
I don't like how it forces people to write `.unwrap()` in correct code (that cannot fail).

It would be possible to implement a `set_texture_filter` which actually changed texture filter. The simple way to implement it is by setting `*self.texture.lock() = None;` which would force the texture to be re-uploaded with the new filter. It is certainly a bit inefficient, but changing filters should be rare and we can optimize it later (by adding `TextureDelta::TextureFilterChanged` or something), and right now it is much nicer than using `Result` and `unwrap` imho.

Reply:
I didn't think about setting `texture = None`, that's a great idea, and much better! :+1:

Pushed an update.